### DISCUSSION TITLE
CombiTimeTable in OM: negative time values

### DISCUSSION THREAD

**Discussion by hillbu1** [OpenModelica/OpenModelica#12782] [General]
Hello,

I have an issue with negative time values in OpenModelica.
We are using the Blocks.Sources.CombiTimeTable to read multiple tables from one txt file as input for our simulations, which works pretty fine, meaning the basic setup and formatting of the table should be correct.
Now I'll need negative time values for a certain reason, which also works basically, meaning the simulation starts at negative time values. The problem is, that the temperatures connected to negative time values are interpreted as 0, i.e., not read correctly. The table looks like this (line starting with two blanks, time and value seperated by tab):
```
double tab_temperature(#lines,2)
  -600	298.15
  0	298.15
  100 273.15
...
```
What happens: Simulation starts at -600 s, but temperature is 0. At time 0 the temperature jumps to 298.15 K, without interpolating from 0 to 298.15. But from here on everything works fine, including interpolation of the temperature between the time values given in the table. The starting time is read from the txt file being the minimum time, which is then set as start time (by surrounding Python code).
So my question is, if there is something wrong with my table or if reading of negative time data is not possible at all, i.e., the first line ( 
 -600\t298.15) isn't read at all and our temperature is simply not initiated resulting in starting temperature = 0.

**Comment by perost** [comment]
I think you need to set the `startTime` parameter in the CombiTimeTable, which by default is 0. The documentation says:
>If time < startTime, no interpolation is performed and the offset is used as ordinate value for all outputs.

**Reply by hillbu1** [reply]
Thanks for your answer! The general starttime of the experiment is set to -600. Do you mean it would need to be set to the same value also in the CombiTimeTable seperately? I would assume (and hope) that this is synchronized within Modelica

**Reply by perost** [reply]
I don't think that's the case unfortunately, the `startTime` parameter in `CombiTimeTable` is just a normal parameter that needs to be set by the user. I tried looking for a reason why it's designed like that, but it's seems to be a case of "it's always been like that and now it's too late to change it".

**Reply by hillbu1** [reply]
Thanks for your hint, it works now! :-)
Actually we used `startTime > 0` already, which worked without any issues.

So now I'll need to check how set this via Python depending on the simulation input or if I can take the quick'n'dirty way (just put `startTime = -1e9` to cover most likely everything we'll ever simulate)
